feat: 채팅방 화면 UI 구현 및 키보드 핸들링 적용

tnemnorivnelee opened the pull request:
## 🔗 관련 이슈

Resolves: #1 

## 📝 작업 내용

- 채팅방 화면 UI 구현 및 키보드 핸들링 적용

## 📷 스크린샷 

### Android

<img width="1080" height="2340" alt="image" src="https://github.com/user-attachments/assets/35036675-a3d7-4e86-9712-33b47c0e5954" />
<img width="1080" height="2340" alt="image" src="https://github.com/user-attachments/assets/6f89f141-68a2-43e2-a8e5-3b85a37a2408" />

---

### IOS

<img width="1170" height="2532" alt="image" src="https://github.com/user-attachments/assets/46e64b02-d901-49e8-80cb-460ccf881705" />

<img width="1170" height="2532" alt="image" src="https://github.com/user-attachments/assets/cd9f8398-d583-485c-81ba-dc2b40784dd3" />



## 💬 리뷰어에게

- 디자인이 잘 반영되었는지 확인 부탁드립니다!

tnemnorivnelee committed:
feat: 채팅 화면 UI 구현 및 키보드 컨트롤러 적용

- ChatHeader, ChatMessage, ChatInput, ChatList 컴포넌트 구현
- 채팅방 네비게이션 분리 (ChatListScreen/ChatScreen)
- react-native-keyboard-controller 적용으로 키보드 처리 개선

tnemnorivnelee committed:
chore: react-native-reanimated 설정 추가

tnemnorivnelee committed:
chore: react-native-worklets-core 설치

tnemnorivnelee committed:
chore: 임시로 reanimated 플러그인 제거

tnemnorivnelee committed:
chore: 임시로 keyboard-controller 제거 (빌드 문제)

tnemnorivnelee committed:
chore: KeyboardStickyView 임시 주석 처리

tnemnorivnelee committed:
chore: KeyboardStickyView 임시 주석 처리

tnemnorivnelee committed:
feat: 채팅 화면 키보드 핸들링 개선

- inverted FlatList + SafeAreaView 기반 채팅 UI로 리팩토링
- 수동 키보드 이벤트 처리로 iOS/Android 키보드 문제 해결
- ChatScreen → ChatRoomScreen 리네임

tnemnorivnelee committed:
style: ChatHeader, ChatInput, ChatMessage 잘못된 디자인 피그마에 맞게 수정

tnemnorivnelee committed:
refactor: iOS 키보드 오프셋 매직 넘버를 상수로 추출

soyeoung1 commented:
채팅 보낸 시간 컬러를 #CED4DB 로 수정하고 현재 코드 디자인에선 채팅 위쪽 모서리 한 부분이 약간 라운드가 되어있는데 아래 사진 피그마 디자인과 같이 아예 라운드 없이 직각으로 되게 수정 가능하시면 부탁드립니다!! 그 외엔 디자인 반영 잘 된 거 같습니다!
<img width="293" height="129" alt="image" src="https://github.com/user-attachments/assets/cb47baa3-2d15-4c13-9fba-91eadbffa406" />

tnemnorivnelee committed:
style: 디자인 토큰 추가 및 적용

tnemnorivnelee committed:
style: 디자인 토큰 구조 개편 및 채팅 UI 적용

tnemnorivnelee committed:
refactor: 디자인 토큰 구조 개편 및 전체 컴포넌트 적용

tnemnorivnelee committed:
refactor: 전역 테마(ThemeProvider) 적용 및 컴포넌트 스타일 마이그레이션

- App 최상단에 ThemeProvider 적용
- 각 컴포넌트의 직접적인 theme import 제거 및 props.theme 사용으로 변경
- emotion.d.ts 타입 정의 추가 및 types 폴더 구조 정리

tnemnorivnelee committed:
chore: emotion.d.ts 파일 복구

tnemnorivnelee committed:
chore: tsconfig.json에 타입 정의 파일(.d.ts) 포함 설정 추가

- 커스텀 타입 정의(emotion.d.ts 등)가 올바르게 인식되도록 include 경로 업데이트

tnemnorivnelee committed:
fix: emotion Theme 인터페이스 확장 방식 수정 및 typeRoots 설정

- emotion.d.ts: type alias 대신 interface extends 방식으로 변경하여 타입 병합 문제 해결
- tsconfig.json: src/types 폴더를 typeRoots에 추가하여 커스텀 타입 인식 보장

tnemnorivnelee commented:
@soyeoung1 
수정한 UI 확인 부탁드립니다!
<img width="1080" height="2340" alt="image" src="https://github.com/user-attachments/assets/6fe4fb11-b2a4-4013-bc32-b0d14c5f22da" />

Yeram9286 commented on src/features/chat/components/ChatMessage.tsx:
Bubble에 border-radius를 16px 말고 13px 정도로 조금 더 뾰족하게 만들면 좋을거같아요

tnemnorivnelee reviewed (commented):
sss

Yeram9286 commented on src/features/chat/components/ChatHeader.tsx:
```diff
@@ -0,0 +1,53 @@
+  background-color: ${({ theme }) => theme.colors.primary.white};
+`;
+
+const LeftButton = styled(TouchableOpacity)`
```
LeftButton 컴포넌트 명을 역할에 맞게 BackButton으로 수정하면 용도를 정확히 알 수있어서 좋을거같아요

Yeram9286 commented on src/features/chat/components/ChatInput.tsx:
```diff
@@ -0,0 +1,70 @@
+  min-height: 48px;
+`;
+
+const StyledInput = styled(TextInput)`
+  flex: 1;
+  font-size: 14px;
+  color: ${({ theme }) => theme.colors.primary.black};
+  max-height: 100px;
+  padding-top: 0;
+  padding-bottom: 0;
+`;
```
<img width="500" height="95" alt="image" src="https://github.com/user-attachments/assets/641ffa26-c63b-4c08-8d00-9652f47bdec3" />
<br>
피그마에 보면 메시지 입력하는 부분이 더 좁은거같아서 min-height: 48px; 부분을 45px 정도로 줄여서 피그마랑 비슷하게 만들면 좋을거같아요

Yeram9286 commented on src/features/chat/components/ChatMessage.tsx:
```diff
@@ -0,0 +1,124 @@
+  margin-bottom: 4px;
+`;
+
+const Bubble = styled(View)<{ isMe: boolean }>`
+  background-color: ${({ isMe, theme }) =>
+    isMe ? theme.colors.primary.main : theme.colors.background.chat};
+  padding: 10px 14px;
+  border-radius: 16px;
+  ${({ isMe }) => (isMe ? 'border-top-right-radius: 0px;' : 'border-top-left-radius: 0px;')}
+`;
```
피그마 보면 buble 부분이 거의 사각에 가까운데, border-radius를 16px 말고 13px 정도로 조금 더 뾰족하게 만들면 좋을거같아요

tnemnorivnelee committed:
refactor: LeftButton -> BackButton 컴포넌트명 변경

tnemnorivnelee committed:
style: 채팅 화면 요소 피그마 수치로 수정

tnemnorivnelee commented:
@Yeram9286 피드백 주신 부분 모두 피그마 수치에 알맞게 수정했습니다!

Yeram9286 reviewed (commented):
!
